pkg/provisioningv2/rke2/planner/certificaterotation.go

```diff
@@ -8,14 +8,24 @@ import (
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
 	"github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1/plan"
 	"github.com/rancher/rancher/pkg/controllers/provisioningv2/rke2"
+	"github.com/sirupsen/logrus"
 )
 
 // rotateCertificates checks if there is a need to rotate any certificates and updates the plan accordingly.
-func (p *Planner) rotateCertificates(controlPlane *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus, clusterPlan *plan.Plan) (rkev1.RKEControlPlaneStatus, error) {
+func (p *Planner) rotateCertificates(controlPlane *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus, tokensSecret plan.Secret, clusterPlan *plan.Plan) (rkev1.RKEControlPlaneStatus, error) {
 	if !shouldRotate(controlPlane) {
 		return status, nil
 	}
 
+	found, joinServer, _, err := p.findInitNode(controlPlane, clusterPlan)
+	if err != nil {
+		logrus.Errorf("[planner] rkecluster %s/%s: error encountered while searching for init node during certificate rotation: %v", controlPlane.Namespace, controlPlane.Name, err)
+	}
+	if !found || joinServer == "" {
+		logrus.Warnf("[planner] rkecluster %s/%s: skipping certificate creation as cluster does not have an init node", controlPlane.Namespace, controlPlane.Name)
+		return status, nil
+	}
```

Comment:
we can just trickle up the error, honestly I don't remember why I made it this way

Reply:
it's tough because we don't have a good feedback mechanism to really determine whether the init node was fixed or not (and I don't really think it makes sense to try and add that at this point).

We can make it error out continously and it'll back off using controller logic. Now that I'm just returning an err, it should properly be trickled onto the status of the controlplane (and then copied back to the prov object)